src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs

```diff
@@ -1660,5 +1660,25 @@ public static void Run()
     }
 }");
         }
+
+        [WorkItem(40644, "https://github.com/dotnet/roslyn/issues/40644")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task ShouldNotWarnForDataMemberFieldsInDataContractClasses()
```

Comment:
💡 Should include a test for a case where a _public_ field of a class marked `[DataContract]` does not include `[DataMember]`. For these cases, `[DataMember]` is optional and the member will be serialized except when it is marked `[IgnoreDataMember]`.

📝 Even though today the test will pass because public fields are excluded from this analysis, it will serve as a regression test for a future scenario where we allow analysis of exposed fields (somehow).

Reply:
➡️ Now fixed